tests/utils/test_metrics.py

```diff
@@ -0,0 +1,170 @@
+from utils.metrics import create_metrics_dict, report_classification, iou
+import torch
+import pytest
+
+# Test arrays: [bs=2, h=2, w,2]
+def init_tensors():
+    pred_multi = torch.tensor([0, 0, 2, 2, 0, 2, 1, 2, 1, 0, 2, 2, 1, 0, 2, 2])
+    pred_binary = torch.tensor([0, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1])
+
+    lbl_multi = torch.tensor([1, 0, 2, 2, 0, 1, 2, 0, 2, 2, 0, 0, 1, 2, 0, 1])
+    lbl_binary = torch.tensor([1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1])
+    # array with dont care
+    lbl_multi_dc = torch.tensor([-1, -1, 2, 2, 0, 1, 2, 0, 2, 2, 0, 0, 1, 2, 0, 1])
+    lbl_binary_dc = torch.tensor([-1, -1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1])
+    return {'pred_multi': pred_multi, 
+            'pred_binary': pred_binary, 
+            'lbl_multi': lbl_multi, 
+            'lbl_binary': lbl_binary, 
+            'lbl_multi_dc': lbl_multi_dc, 
+            'lbl_binary_dc': lbl_binary_dc}
+
+
+class TestMetrics(object):
+    def test_create_metrics_dict(self):
+        """Evaluate the metrics dictionnary creation. 
+        Binary and multiclass"""
+        # binary tasks have 1 class at class definition.
+        num_classes = 1
+        metrics_dict = create_metrics_dict(num_classes)
+        assert 'iou_1' in metrics_dict.keys()
+        assert 'iou_2' not in metrics_dict.keys()
+
+        num_classes = 3
+        metrics_dict = create_metrics_dict(num_classes)
+        assert 'iou_1' in metrics_dict.keys()
+        assert 'iou_2' in metrics_dict.keys()
+        assert 'iou_3' not in metrics_dict.keys()
+        del metrics_dict
+
+    def test_report_classification_multi(self):
+        """Evaluate report classification. 
+        Multiclass, without ignore_index in array."""
+        t = init_tensors()
+        metrics_dict = create_metrics_dict(3)
+        metrics_dict = report_classification(t['pred_multi'], 
+                                             t['lbl_multi'], 
+                                             batch_size=2, 
+                                             metrics_dict=metrics_dict,
+                                             ignore_index=-1)
+        assert "{:.6f}".format(metrics_dict['precision'].val) == "0.327083"
```

Comment:
Do we want a log of this in case of test on others machines like HPC?

Reply:
I don't see the necessity of logs for unit tests. Pytest does output a pretty detailed log if 
tests fail...